src/nunit.gui.core/Model/TestModel.cs

```diff
@@ -96,32 +96,30 @@ public bool HasResults
 
         private IList<string> _files;
 
-        // TODO: We are directly using an engine class here rather
-        // than going through the API - need to fix this.
-        // Note that the engine returns more values than we really
-        // want to list. For example, we don't distinguish between
-        // client and full profiles when executing tests. So we do
-        // a lot of manipulation of this list here. Even if some of
-        // the problems in the engine are fixed, we may have to
-        // retain this code in order to work with older engines.
-        private List<RuntimeFramework> _runtimes;
-        public IList<RuntimeFramework> AvailableRuntimes
+        // The engine returns more values than we really want.
+        // For example, we don't currently distinguish between
+        // client and full profiles when executing tests. We
+        // drop unwanted entries here. Even if some of these items
+        // are removed in a later version of the engine, we may
+        // have to retain this code to work with older engines.
+        private List<IRuntimeFramework> _runtimes;
+        public IList<IRuntimeFramework> AvailableRuntimes
         {
             get
             {
                 if (_runtimes == null)
                 {
-                    _runtimes = new List<RuntimeFramework>();
-                    foreach (var runtime in RuntimeFramework.AvailableFrameworks)
+                    _runtimes = new List<IRuntimeFramework>();
+                    var availableFrameworks = GetService<IAvailableRuntimes>();
+                    foreach (var runtime in GetService<IAvailableRuntimes>().AvailableRuntimes)
                     {
                         // Nothing below 2.0
                         if (runtime.ClrVersion.Major < 2)
                             continue;
 
                         // Remove erroneous entries for 4.5 Client profile
-                        if (runtime.FrameworkVersion.Major == 4 && runtime.FrameworkVersion.Minor > 0)
-                            if (runtime.DisplayName.EndsWith("Client"))
-                                continue;
+                        if (runtime.FrameworkVersion.Major == 4 && runtime.FrameworkVersion.Minor > 0 && runtime.Profile == "Client")
```

Comment:
I feel like for 3 conditions it should be moved to a small method like `if (IsFrameworkErronious(runtime))`. I know the comment exists above it but my eyes glaze over comments usually.

Reply:
OK